moveit_ros/move_group/src/default_capabilities/get_group_urdf_capability.cpp

```diff
@@ -62,6 +62,10 @@ GetUrdfService::GetUrdfService() : MoveGroupCapability("get_group_urdf")
 
 void GetUrdfService::initialize()
 {
+  robot_description_subscriber_ = context_->moveit_cpp_->getNode()->create_subscription<std_msgs::msg::String>(
```

Comment:
Should we make sure the topic `robot_description` is available and wait for a given amount of time if not?

Reply:
I tried using [this rclcpp function](https://github.com/ros2/rclcpp/blob/foxy/rclcpp/include/rclcpp/wait_for_message.hpp#L39-L44) to wait for a certain time until we get a message in the `robot_description` topic. But this causes a timeout error when using this service. Not sure I'm using it wrong.

I could add a while loop with a timer to check if the `full_urdf_string_` variable is set. 

Do you have a better solution to this?

Reply:
I thought about this a bit more and because you have a moveit_cpp you should be able to get the URDF string like this and not need to manually handle the subscription.

If you add a function to RDFLoader to get the `urdf_string_` value it contains. Maybe name it `getUrdfString() -> std::string`. Then you can do this:

```context_->moveit_cpp_->getPlanningSceneMonitor()->getRobotModelLoader()->getRDFLoader()->getUrdfString();```

I know it is ugly but if you have a moveit_cpp and it is initialized you should be able to get the string this way too.